tests/assemble.py

```diff
@@ -121,6 +121,20 @@ def test_assemble_global():
     assert exported_symbols == [(0, 'counter'), (1, 'internal'), (2, 'entry')]
 
 
+def test_assemble_uppercase_opcode():
+    a = Assembler()
+    try:
+        a.assemble("  WAIT 42")
+    except Exception as e:
+        if str(e) != "Unknown opcode or directive: WAIT":
+            # re-raise failures we didn't expect
+            raise
+        raised = True
+    else:
+        raised = False
+    assert not raised
```

Comment:
see `pytest.raises`.

Reply:
Do you suggest to add a dependency on `pytest` or to extract the logic into a helper method that is similar to `pytest.raises`? (The latter might be quite nice because I did end up checking for exceptions more than a few times)

Anyway, I followed the pattern that was used in the existing tests, because that worked and used the tooling available.

I'm also not aware of any `pytest` module for MicroPython, and the `unittest` lib that lives in `micropython-lib` is a bit broken. So I was happy with your simple/pragmatic approach to testing for raised exception and duplicated the approach.

Do you feel we should use the opportunity here to improve the testing tooling/framworks used?

Reply:
oops, no, we better don't add pytest.

i'm used to it, but maybe better not for micropython.